### PR TITLE
Add xml group tests

### PR DESCRIPTION
**Pull request by zhfeng**

### CODE REVIEW

**Review by jamesnetherton — APPROVED** [review]

File: product/src/main/resources/camel-quarkus-product-source.json [hunk 1]
```diff
@@ -620,6 +620,7 @@
             "includes": [
                 "integration-tests/*/pom.xml",
                 "integration-test-groups/*/*/pom.xml",
+                "integration-test-groups/*/*/*/pom.xml",
```
Comment (zhfeng) [inline product/src/main/resources/camel-quarkus-product-source.json]: @ppalaga I just wonder if we can use `integration-test-groups/**/pom.xml` to includes all of the group tests?

Comment (ppalaga) [reply product/src/main/resources/camel-quarkus-product-source.json]: The `**` wildcard is definitely supported, but we want to include only modules that can be executed as tests, esp. not the individual group parents, like `integration-test-groups/xml/pom.xml` or `integration-test-groups/http/pom.xml`. Hence I think what could work is the following: 

```json
            "includes": [
                "integration-tests/*/pom.xml",
                "integration-test-groups/*/**/pom.xml",
                "integration-tests-jvm/*/pom.xml"
            ],
```

Comment (zhfeng) [reply product/src/main/resources/camel-quarkus-product-source.json]: OK, but `"integration-test-groups/*/**/pom.xml"` still includes `integration-test-groups/xml/pom.xml`. So it has to add `"integration-test-groups/*/pom.xml"` in the `excludes`.

**Comment by zhfeng** [comment]
Hmm, when I run `./mvnw -f integration-test-groups/xml/native/xslt-classpath/pom.xml clean test -Pprod`, it is failing with
```
[INFO] Scanning for projects...
[INFO] Loaded 7 groupId for remote repository atlassian
[INFO] 
[INFO] --< org.apache.camel.quarkus:camel-quarkus-integration-test-xslt-classpath >--
[INFO] Building Camel Quarkus :: Integration Tests :: Xslt Classpath 3.8.0.fuse-SNAPSHOT
[INFO]   from pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
[WARNING] The POM for org.apache.camel.quarkus:camel-quarkus-stax-deployment:pom:3.8.0.fuse-SNAPSHOT is missing, no dependency information available
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.304 s
[INFO] Finished at: 2024-05-14T15:04:43+08:00

```

`camel-quarkus-stax` is a community supported extension. So `camel-quarkus-stax-deployment` should be with the community version. @ppalaga How can we handle this by `cq-maven-plugin` or is there any other way?

File: product/src/main/resources/camel-quarkus-product-source.json (hunk 1 above)
Comment (ppalaga) [reply product/src/main/resources/camel-quarkus-product-source.json]: > OK, but `"integration-test-groups/*/**/pom.xml"` still includes `integration-test-groups/xml/pom.xml`. So it has to add `"integration-test-groups/*/pom.xml"` in the `excludes`.

Sorry about that. With the exclude, we are on the same amount of lines. Feel free to decide which you find more elegant.

**Comment by ppalaga** [comment]
@zhfeng TLDR: just add `-DnoVirtualDependencies` to your Maven command.

Long version: this happens due to [these -deployment dependencies](https://github.com/jboss-fuse/camel-quarkus/blob/3.8.0-product/integration-test-groups/xml/native/xslt-classpath/pom.xml#L124-L216) that we add so that they are built before the test during the build of the whole source tree. Those use `${project.version}` rather than the managed version and therefore they have to be present in the local Maven repo before you run the single test module with `mvn -f ...` **UNLESS** you disable the `virtualDependencies` profile with `-DnoVirtualDependencies`.

**Pull request merged**